[FIX] Create Class: Remove context settings in favour of schema-only non-context

janezd opened the pull request:
##### Issue

Fixes #5499, assuming that weird behaviour is caused by context settings. (Which it almost certainly is.)

##### Description of changes

As per title, I replaced context settings with schema-only (non-context) settings. The only remaining context setting is the variable.

It gets even better. Rules were already an "internal context setting" with the variable as a context. That is, if the user sets the rules for one variable, switches to another, sets some rules, and switches back (s)he will get the rules for the former variable.

After removing context settings, this works as hinting. Load zoo, set some rules for type. Load Iris (rules disappear), set different rules. Go back to zoo and you get the rules back. This hinting is based on variable name.

If this is merged and documentation is updated, close #6961, too.

##### Includes
- [X] Code changes
- [X] Tests

janezd committed:
Create Class: Remove context settings in favour of schema-only non-context

janezd committed:
Create Class: Improve layout

markotoplak committed:
owcreateclass: also migrate atttribute and regular expressions

markotoplak commented:
@janezd, I (think I) fixed settings migration. You forgot to migrate the attribute (which stays in the context and regular expression toggle.

I do not quite understand what you wanted to do with `update_dynamic_height`. Now it is only used at init (it has another unused parameter), and for me, what I see it doing, is to make the widget wider. I never missed width here. Or what was the reasoning behind it?

janezd committed:
Create Class: Fix resizing when removing rules, tighten the layout

janezd commented:
`update_dynamic_height` was likely an attempt to reduce the height of the widget when the user removes a rule. It probably didn't work and I removed the call from `remove_row`, but forgot to remove the entire function and the call from `__init__`.

Now I removed the function and set the flag that will resize the widget automatically.
